src/main/kotlin/com/sourcegraph/cody/history/HistoryTree.kt

```diff
@@ -1,84 +1,125 @@
 package com.sourcegraph.cody.history
 
-import com.intellij.ui.treeStructure.Tree
-import com.sourcegraph.cody.history.listener.DeleteListener
-import com.sourcegraph.cody.history.listener.DoubleLeftClickListener
-import com.sourcegraph.cody.history.listener.EnterListener
-import com.sourcegraph.cody.history.listener.RightClickListener
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.ui.SimpleToolWindowPanel
+import com.intellij.ui.PopupHandler
+import com.intellij.ui.ScrollPaneFactory
+import com.intellij.ui.treeStructure.SimpleTree
+import com.intellij.util.EditSourceOnDoubleClickHandler
 import com.sourcegraph.cody.history.state.ChatState
-import com.sourcegraph.cody.history.util.ToolbarDurationTextFormatter
-import java.awt.event.MouseEvent
-import java.time.LocalDateTime
-import java.time.temporal.ChronoUnit
+import com.sourcegraph.cody.history.util.DurationGroupFormatter
+import java.awt.event.ActionEvent
+import java.awt.event.KeyEvent
+import javax.swing.*
 import javax.swing.tree.DefaultMutableTreeNode
 import javax.swing.tree.DefaultTreeModel
+import javax.swing.tree.TreeSelectionModel
 
-class HistoryTree(onSelect: (ChatState) -> Unit, onDelete: (ChatState) -> Unit) : Tree() {
+class HistoryTree(
+    private val onSelect: (ChatState) -> Unit,
+    private val onDelete: (ChatState) -> Unit
+) : SimpleToolWindowPanel(true, true) {
 
-  private val popup = HistoryPopupMenu(onSelect, onDelete, getSelected = { getLeafOrNull()!!.chat })
```

Comment:
What are those todo commit msgs?

Reply:
Oh, I forgot to remove them.

Thanks! Fixed.